add-bookings

kpak292 opened the pull request.

kpak292 committed:
1) deleted InMemoryItemRepository.java and InMemoryUserRepository.java
2) Realized JpaRepositories

kpak292 committed:
changed Item
UserId -> User (ManyToOne)

kpak292 committed:
Realized
1) Booking API
2) Comments API

kpak292 committed:
Realized
1) Booking API
2) Comments API

kpak292 committed:
Merge remote-tracking branch 'origin/add-bookings' into add-bookings

# Conflicts:
#	src/main/java/ru/practicum/shareit/booking/dto/exceptions/NotAvailableException.java
#	src/main/java/ru/practicum/shareit/booking/entity/Booking.java
#	src/main/java/ru/practicum/shareit/booking/repository/BookingRepository.java
#	src/main/java/ru/practicum/shareit/booking/service/BookingService.java
#	src/main/java/ru/practicum/shareit/booking/service/BookingServiceImpl.java
#	src/main/java/ru/practicum/shareit/exceptions/ErrorHandler.java
#	src/main/java/ru/practicum/shareit/item/controller/ItemController.java
#	src/main/java/ru/practicum/shareit/item/service/ItemServiceImpl.java

kpak292 committed:
Fix

kpak292 committed:
Fix Style

kpak292 committed:
deleted tests

kpak292 committed:
Deleted parameter
spring.jackson.default-property-inclusion = NON_NULL

kpak292 committed:
Commented check for user existance in ItemServiceImpl.java

bismuthD commented on src/main/java/ru/practicum/shareit/booking/dto/BookingMapper.java:
```diff
@@ -0,0 +1,18 @@
+
+import java.util.Collection;
+
+@Mapper
```
верно использовал аннотацию, так же для классов-мапперов можно на альтернативу использовать аннотацию @NoArgsConstructor(access = AccessLevel.PRIVATE)

kpak292 replied:
Данная аннотация не работает в интерфейсе, только для классов и Enum. Пока оставил как есть

bismuthD commented on src/main/java/ru/practicum/shareit/booking/dto/BookingDto.java:
```diff
@@ -1,7 +1,48 @@
+    ItemDto item;
+
+    @NotNull(message = "Start cannot be null")
+    @Future(message = "Start date cannot be in past")
```
тут следует использовать аннотацию @FutureOrPresent

kpak292 replied:
Поменял

bismuthD commented on src/main/java/ru/practicum/shareit/booking/dto/BookingDto.java:
```diff
@@ -1,7 +1,48 @@
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@AllArgsConstructor
+@NoArgsConstructor
 public class BookingDto {
```
Хорошо было бы ограничить размеры текстовых полей на основе тех ограничений, которые установлены на уровне таблицы в БД, чтобы заранее откинуть некорректные запросы. Для ограничения количества символов можно использовать @Size. Эта аннотация удобна тем, что допускает null, за счет чего при обновлении по значению мы сможем понять, было передано значение или нет

kpak292 replied:
Именно в этом классе нет текстовых полей, однако поменял во всех DTO

bismuthD reviewed (commented):
Отличная работа, оставил несколько рекомендаций для улучшения кода

kpak292 committed:
1) Optimized loadCommentsAndBookings in ItemServiceImpl.java
2) Added @Size to all strings in DTOs
3) Changed Annotation @Future to @FutureOrPresent in BookingDto

kpak292 committed:
deleted unused imports

kpak292 committed:
deleted unused imports

kpak292 commented:
Странная штука, поменял аннотацию @Future на @FutureOrPresent и постман тесты дали ошибку, еще раз дал пустой коммит (комментарий) и все прошло.
На локальной машине воспроизводится, когда не срабатывают wait
